{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          unique_value  count\n",
      "0     NameError(\"name 'oppdater_matvare' is not def...    669\n",
      "1       NameError(\"name 'finn_pris' is not defined\")]]    479\n",
      "2                          AssertionError('0 != 99')]]    284\n",
      "3              IndexError('list index out of range')]]    266\n",
      "4            IndexError('string index out of range')]]    231\n",
      "..                                                 ...    ...\n",
      "594   AttributeError(\"'NoneType' object has no attr...      1\n",
      "595       NameError(\"name 'bokstav' is not defined\")]]      1\n",
      "596             NameError(\"name 'S' is not defined\")]]      1\n",
      "597                      AssertionError(\"'c' != []\")]]      1\n",
      "598     NameError(\"name 'vispriser' is not defined\")]]      1\n",
      "\n",
      "[599 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H21/errorH21.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "unique_values_df = unique_values.rename_axis('unique_value').reset_index(name='count')\n",
    "\n",
    "# Display the grouped data as a table\n",
    "print(unique_values_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sum of NameError:  45\n",
      "Sum of AssertionError:  1724\n",
      "Sum of TypeError:  611\n",
      "Sum of IndexError:  510\n",
      "Sum of AttributeError:  232\n",
      "Sum of UnboundLocalError:  57\n",
      "Sum of ValueError:  83\n",
      "Sum of KeyError:  45\n"
     ]
    }
   ],
   "source": [
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H21/errorH21.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "# Filter unique values containing \"NameError\"\n",
    "name_error_values = [value for value in unique_values.index if \"KeyError\" in value]\n",
    "name_error_counts = unique_values[name_error_values]\n",
    "\n",
    "# Sum the counts of the NameError values\n",
    "name_error_sum = name_error_counts.sum()\n",
    "\n",
    "# Filter unique values containing \"AssertionError\"\n",
    "assertion_error_values = [value for value in unique_values.index if \"AssertionError\" in value]\n",
    "assertion_error_counts = unique_values[assertion_error_values]\n",
    "\n",
    "# Sum the counts of the AssertionError values\n",
    "assertion_error_sum = assertion_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"TypeError\"\n",
    "type_error_values = [value for value in unique_values.index if \"TypeError\" in value]\n",
    "type_error_counts = unique_values[type_error_values]\n",
    "\n",
    "# Sum the counts of the TypeError values\n",
    "type_error_sum = type_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"IndexError\"\n",
    "index_error_values = [value for value in unique_values.index if \"IndexError\" in value]\n",
    "index_error_counts = unique_values[index_error_values]\n",
    "\n",
    "# Sum the counts of the IndexError values\n",
    "index_error_sum = index_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"AttributeError\"\n",
    "attribute_error_values = [value for value in unique_values.index if \"AttributeError\" in value]\n",
    "attribute_error_counts = unique_values[attribute_error_values]\n",
    "\n",
    "# Sum the counts of the AttributeError values\n",
    "attribute_error_sum = attribute_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"UnboundLocalError\"\n",
    "unbound_local_error_values = [value for value in unique_values.index if \"UnboundLocalError\" in value]\n",
    "unbound_local_error_counts = unique_values[unbound_local_error_values]\n",
    "\n",
    "# Sum the counts of the UnboundLocalError values\n",
    "unbound_local_error_sum = unbound_local_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"ValueError\"\n",
    "value_error_values = [value for value in unique_values.index if \"ValueError\" in value]\n",
    "value_error_counts = unique_values[value_error_values]\n",
    "\n",
    "# Sum the counts of the ValueError values\n",
    "value_error_sum = value_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"KeyError\"\n",
    "key_error_values = [value for value in unique_values.index if \"KeyError\" in value]\n",
    "key_error_counts = unique_values[key_error_values]\n",
    "\n",
    "# Sum the counts of the KeyError values\n",
    "key_error_sum = key_error_counts.sum()\n",
    "\n",
    "# Display the grouped data as a table\n",
    "print(\"Sum of NameError: \", name_error_sum)\n",
    "print(\"Sum of AssertionError: \", assertion_error_sum)\n",
    "print(\"Sum of TypeError: \", type_error_sum)\n",
    "print(\"Sum of IndexError: \", index_error_sum)\n",
    "print(\"Sum of AttributeError: \", attribute_error_sum)\n",
    "print(\"Sum of UnboundLocalError: \", unbound_local_error_sum)\n",
    "print(\"Sum of ValueError: \", value_error_sum)\n",
    "print(\"Sum of KeyError: \", key_error_sum)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          unique_value  count\n",
      "0     NameError(\"name 'oppdater_matvare' is not def...    669\n",
      "1       NameError(\"name 'finn_pris' is not defined\")]]    479\n",
      "2                          AssertionError('0 != 99')]]    284\n",
      "3              IndexError('list index out of range')]]    266\n",
      "4            IndexError('string index out of range')]]    231\n",
      "..                                                 ...    ...\n",
      "594   AttributeError(\"'NoneType' object has no attr...      1\n",
      "595       NameError(\"name 'bokstav' is not defined\")]]      1\n",
      "596             NameError(\"name 'S' is not defined\")]]      1\n",
      "597                      AssertionError(\"'c' != []\")]]      1\n",
      "598     NameError(\"name 'vispriser' is not defined\")]]      1\n",
      "\n",
      "[599 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H21/errorH21.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "unique_values_df = unique_values.rename_axis('unique_value').reset_index(name='count')\n",
    "\n",
    "# Display the grouped data as a table\n",
    "print(unique_values_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          unique_value  count\n",
      "0        AssertionError('False is not true : False')]]    554\n",
      "1              IndexError('list index out of range')]]     50\n",
      "2         AssertionError('False is not true : True')]]     47\n",
      "3          NameError(\"name 'lenght' is not defined\")]]     43\n",
      "4              NameError(\"name 'np' is not defined\")]]     41\n",
      "..                                                 ...    ...\n",
      "207       NameError(\"name 'lst_sum' is not defined\")]]      1\n",
      "208                        AssertionError('5 != 11')]]      1\n",
      "209   UnboundLocalError(\"local variable 'restulat' ...      1\n",
      "210                      AssertionError('-11 != 11')]]      1\n",
      "211         NameError(\"name 'ragne' is not defined\")]]      1\n",
      "\n",
      "[212 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H22/errorH22morgen.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "unique_values_df = unique_values.rename_axis('unique_value').reset_index(name='count')\n",
    "\n",
    "# Display the grouped data as a table\n",
    "print(unique_values_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sum of NameError:  0\n",
      "Sum of AssertionError:  814\n",
      "Sum of TypeError:  149\n",
      "Sum of IndexError:  181\n",
      "Sum of AttributeError:  41\n",
      "Sum of UnboundLocalError:  31\n",
      "Sum of ValueError:  46\n",
      "Sum of KeyError:  0\n"
     ]
    }
   ],
   "source": [
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H22/errorH22morgen.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "# Filter unique values containing \"NameError\"\n",
    "name_error_values = [value for value in unique_values.index if \"KeyError\" in value]\n",
    "name_error_counts = unique_values[name_error_values]\n",
    "\n",
    "# Sum the counts of the NameError values\n",
    "name_error_sum = name_error_counts.sum()\n",
    "\n",
    "# Filter unique values containing \"AssertionError\"\n",
    "assertion_error_values = [value for value in unique_values.index if \"AssertionError\" in value]\n",
    "assertion_error_counts = unique_values[assertion_error_values]\n",
    "\n",
    "# Sum the counts of the AssertionError values\n",
    "assertion_error_sum = assertion_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"TypeError\"\n",
    "type_error_values = [value for value in unique_values.index if \"TypeError\" in value]\n",
    "type_error_counts = unique_values[type_error_values]\n",
    "\n",
    "# Sum the counts of the TypeError values\n",
    "type_error_sum = type_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"IndexError\"\n",
    "index_error_values = [value for value in unique_values.index if \"IndexError\" in value]\n",
    "index_error_counts = unique_values[index_error_values]\n",
    "\n",
    "# Sum the counts of the IndexError values\n",
    "index_error_sum = index_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"AttributeError\"\n",
    "attribute_error_values = [value for value in unique_values.index if \"AttributeError\" in value]\n",
    "attribute_error_counts = unique_values[attribute_error_values]\n",
    "\n",
    "# Sum the counts of the AttributeError values\n",
    "attribute_error_sum = attribute_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"UnboundLocalError\"\n",
    "unbound_local_error_values = [value for value in unique_values.index if \"UnboundLocalError\" in value]\n",
    "unbound_local_error_counts = unique_values[unbound_local_error_values]\n",
    "\n",
    "# Sum the counts of the UnboundLocalError values\n",
    "unbound_local_error_sum = unbound_local_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"ValueError\"\n",
    "value_error_values = [value for value in unique_values.index if \"ValueError\" in value]\n",
    "value_error_counts = unique_values[value_error_values]\n",
    "\n",
    "# Sum the counts of the ValueError values\n",
    "value_error_sum = value_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"KeyError\"\n",
    "key_error_values = [value for value in unique_values.index if \"KeyError\" in value]\n",
    "key_error_counts = unique_values[key_error_values]\n",
    "\n",
    "# Sum the counts of the KeyError values\n",
    "key_error_sum = key_error_counts.sum()\n",
    "\n",
    "# Display the grouped data as a table\n",
    "print(\"Sum of NameError: \", name_error_sum)\n",
    "print(\"Sum of AssertionError: \", assertion_error_sum)\n",
    "print(\"Sum of TypeError: \", type_error_sum)\n",
    "print(\"Sum of IndexError: \", index_error_sum)\n",
    "print(\"Sum of AttributeError: \", attribute_error_sum)\n",
    "print(\"Sum of UnboundLocalError: \", unbound_local_error_sum)\n",
    "print(\"Sum of ValueError: \", value_error_sum)\n",
    "print(\"Sum of KeyError: \", key_error_sum)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          unique_value  count\n",
      "0        AssertionError('False is not true : False')]]    554\n",
      "1              IndexError('list index out of range')]]     50\n",
      "2         AssertionError('False is not true : True')]]     47\n",
      "3          NameError(\"name 'lenght' is not defined\")]]     43\n",
      "4              NameError(\"name 'np' is not defined\")]]     41\n",
      "..                                                 ...    ...\n",
      "207       NameError(\"name 'lst_sum' is not defined\")]]      1\n",
      "208                        AssertionError('5 != 11')]]      1\n",
      "209   UnboundLocalError(\"local variable 'restulat' ...      1\n",
      "210                      AssertionError('-11 != 11')]]      1\n",
      "211         NameError(\"name 'ragne' is not defined\")]]      1\n",
      "\n",
      "[212 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H22/errorH22morgen.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "unique_values_df = unique_values.rename_axis('unique_value').reset_index(name='count')\n",
    "\n",
    "# Display the grouped data as a table\n",
    "print(unique_values_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         unique_value  count\n",
      "0    TypeError('array() takes from 1 to 2 position...    125\n",
      "1                         AssertionError('0 != 15')]]     20\n",
      "2             IndexError('list index out of range')]]     19\n",
      "3           AssertionError('None is not true : M3')]]     12\n",
      "4    IndexError('arrays used as indices must be of...     11\n",
      "..                                                ...    ...\n",
      "75         AssertionError('0.0 is not true : 38.2')]]      1\n",
      "76       NameError(\"name 'ndarray' is not defined\")]]      1\n",
      "77   TypeError('list.append() takes exactly one ar...      1\n",
      "78   AttributeError(\"module 'numpy' has no attribu...      1\n",
      "79       NameError(\"name 'heltall' is not defined\")]]      1\n",
      "\n",
      "[80 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H22Kveld/errorH22kveld.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "unique_values_df = unique_values.rename_axis('unique_value').reset_index(name='count')\n",
    "\n",
    "# Display the grouped data as a table\n",
    "print(unique_values_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sum of NameError:  0\n",
      "Sum of AssertionError:  63\n",
      "Sum of TypeError:  181\n",
      "Sum of IndexError:  41\n",
      "Sum of AttributeError:  11\n",
      "Sum of UnboundLocalError:  14\n",
      "Sum of ValueError:  7\n",
      "Sum of KeyError:  0\n"
     ]
    }
   ],
   "source": [
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H22Kveld/errorH22kveld.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "# Filter unique values containing \"NameError\"\n",
    "name_error_values = [value for value in unique_values.index if \"KeyError\" in value]\n",
    "name_error_counts = unique_values[name_error_values]\n",
    "\n",
    "# Sum the counts of the NameError values\n",
    "name_error_sum = name_error_counts.sum()\n",
    "\n",
    "# Filter unique values containing \"AssertionError\"\n",
    "assertion_error_values = [value for value in unique_values.index if \"AssertionError\" in value]\n",
    "assertion_error_counts = unique_values[assertion_error_values]\n",
    "\n",
    "# Sum the counts of the AssertionError values\n",
    "assertion_error_sum = assertion_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"TypeError\"\n",
    "type_error_values = [value for value in unique_values.index if \"TypeError\" in value]\n",
    "type_error_counts = unique_values[type_error_values]\n",
    "\n",
    "# Sum the counts of the TypeError values\n",
    "type_error_sum = type_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"IndexError\"\n",
    "index_error_values = [value for value in unique_values.index if \"IndexError\" in value]\n",
    "index_error_counts = unique_values[index_error_values]\n",
    "\n",
    "# Sum the counts of the IndexError values\n",
    "index_error_sum = index_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"AttributeError\"\n",
    "attribute_error_values = [value for value in unique_values.index if \"AttributeError\" in value]\n",
    "attribute_error_counts = unique_values[attribute_error_values]\n",
    "\n",
    "# Sum the counts of the AttributeError values\n",
    "attribute_error_sum = attribute_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"UnboundLocalError\"\n",
    "unbound_local_error_values = [value for value in unique_values.index if \"UnboundLocalError\" in value]\n",
    "unbound_local_error_counts = unique_values[unbound_local_error_values]\n",
    "\n",
    "# Sum the counts of the UnboundLocalError values\n",
    "unbound_local_error_sum = unbound_local_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"ValueError\"\n",
    "value_error_values = [value for value in unique_values.index if \"ValueError\" in value]\n",
    "value_error_counts = unique_values[value_error_values]\n",
    "\n",
    "# Sum the counts of the ValueError values\n",
    "value_error_sum = value_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"KeyError\"\n",
    "key_error_values = [value for value in unique_values.index if \"KeyError\" in value]\n",
    "key_error_counts = unique_values[key_error_values]\n",
    "\n",
    "# Sum the counts of the KeyError values\n",
    "key_error_sum = key_error_counts.sum()\n",
    "\n",
    "# Display the grouped data as a table\n",
    "print(\"Sum of NameError: \", name_error_sum)\n",
    "print(\"Sum of AssertionError: \", assertion_error_sum)\n",
    "print(\"Sum of TypeError: \", type_error_sum)\n",
    "print(\"Sum of IndexError: \", index_error_sum)\n",
    "print(\"Sum of AttributeError: \", attribute_error_sum)\n",
    "print(\"Sum of UnboundLocalError: \", unbound_local_error_sum)\n",
    "print(\"Sum of ValueError: \", value_error_sum)\n",
    "print(\"Sum of KeyError: \", key_error_sum)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         unique_value  count\n",
      "0    TypeError('array() takes from 1 to 2 position...    125\n",
      "1                         AssertionError('0 != 15')]]     20\n",
      "2             IndexError('list index out of range')]]     19\n",
      "3           AssertionError('None is not true : M3')]]     12\n",
      "4    IndexError('arrays used as indices must be of...     11\n",
      "..                                                ...    ...\n",
      "75         AssertionError('0.0 is not true : 38.2')]]      1\n",
      "76       NameError(\"name 'ndarray' is not defined\")]]      1\n",
      "77   TypeError('list.append() takes exactly one ar...      1\n",
      "78   AttributeError(\"module 'numpy' has no attribu...      1\n",
      "79       NameError(\"name 'heltall' is not defined\")]]      1\n",
      "\n",
      "[80 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H22Kveld/errorH22kveld.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "unique_values_df = unique_values.rename_axis('unique_value').reset_index(name='count')\n",
    "\n",
    "# Display the grouped data as a table\n",
    "print(unique_values_df)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
